src/openrct2/localisation/Formatting.cpp

```diff
@@ -295,14 +295,17 @@ namespace OpenRCT2
         }
     }
 
-    template<size_t TSize, typename TIndex> static void AppendSeparator(char (&buffer)[TSize], TIndex& i, std::string_view sep)
+    template<size_t TSize, typename TIndex>
+    static void AppendSeparatorReversed(char (&buffer)[TSize], TIndex& i, std::string_view sep)
     {
-        if (i < TSize)
+        if (i + sep.size() >= TSize)
+            return;
```

Comment:
Shouldn't it return before as well, e.g. `i >= TSize`? Otherwise, you could still end up with half a character near the end of the buffer?

Reply:
I don’t understand, that’s what it’s doing already?

Reply:
What if the `TIndex& i` parameter is already indexing beyond the buffer size `size_t TSize`, though?

Reply:
I really don’t understand what you’re trying to say. If `i` is equal to or bigger than `TSize`, then `i + sep.size()` is also bigger than `TSize`. So it will return. Are you sure you’re reading this properly?

Reply:
Oh, you're right. I can see what I was thinking -- it would have been a problem were this a `<` comparison. Clearly, not an issue here, though.